docs/using_tensorgraph.md

```diff
@@ -0,0 +1,101 @@
+# Using TensorGraph
```

Comment:
The introductory material here looks good!

We should add a citation to Keras (or just credit them for the design informally) since we borrowed a lot of the stylistic form from them. It might also be useful to have a "how is this not keras" section that explains the differing semantics of `Layer` objects. This is more for advanced users, but I anticipate we'll get at least a few people who have tried Keras before trying the new Tensorgraphs.